Add full support for logging as an alternative to progress bars

opcode81 opened the pull request:
Add full support for logging as an alternative to progress bars printed to stderr

  * Add configuration items * use_pbar (use progress bar instead of logging) * log_level (level to use when logging is enabled/progress bar is disabled)
  * Allow to override use_pbar for every push/pull operation individually
  * Show readable file sizes (with adaptive units) in log messages

Resolves #31 #25

opcode81 committed:
Add full support for logging as an alternative to progress bars printed to stderr

  * Add configuration items
      * use_pbar (use progress bar instead of logging)
      * log_level (level to use when logging is enabled/progress bar is disabled)
  * Allow to override use_pbar for every push/pull operation individually
  * Show readable file sizes (with adaptive units) in log messages

MischaPanch commented:
Seems like a github action was deprecated leading to the tests failing. I will fix this on master, which you can then merge into this branch

MischaPanch commented on src/accsr/remote_storage.py:
```diff
@@ -320,12 +320,16 @@ def size_files_to_sync(self) -> int:
-            )
-        mode = "local" if self.sync_direction == "push" else "remote"
-        return _get_total_size(self.files_to_sync, mode=mode)
+        assert self.sync_direction is not None
```
Pls add the message from the runtime error here

opcode81 replied:
My recent update which simplified the interfaces has removed the need for this assertion.

opcode81 replied:
Please review this commit: https://github.com/appliedAI-Initiative/accsr/pull/32/commits/1570c0f58f1dbd03aa2a833263dd8abde25f67d2

MischaPanch replied:
Nice!

MischaPanch commented on src/accsr/remote_storage.py:
```diff
@@ -512,7 +525,12 @@ def driver(self) -> StorageDriver:
+        file_count: Tuple[int, int],
+        direction: Literal["push", "pull"],
+        force=False,
+        use_pbar: Optional[bool] = None,
```
docstring missing. Will this override `self.use_pbar`?

opcode81 replied:
Added

MischaPanch commented on src/accsr/remote_storage.py:
```diff
@@ -648,8 +682,39 @@ def _listed_due_to_name_collision(
+            log.log(self.conf.log_level, msg)
+
+    @staticmethod
+    def _readable_size(num_bytes: int) -> str:
```
Nice one!

opcode81 committed:
Add sync_direction to SyncObject, simplifying access to transferred bytes/size

This removes the option to pass a string instead of a SyncObject to TransactionSummary.add_entry

opcode81 committed:
Use unit scale in progress bar

MischaPanch commented:
Merging, will fix the GH action on  master and if all goes well release the next version of accsr
